FinSearchUnified/Tests/Helper/StaticHelperTest.php

```diff
@@ -797,4 +812,73 @@ public function priceFilterProvider()
             ]
         ];
     }
+
+    /**
+     * @dataProvider directIntegrationProvider
+     *
+     * @param string $integrationType
+     * @param string $expectedResult
+     * @param array $directIntegration
+     *
+     * @throws Exception
+     */
+    public function testDirectIntegration($integrationType, $expectedResult, $directIntegration)
+    {
+        // Create Mock object for Shopware Config
+        $mockConfig = $this->getMockBuilder(Config::class)
+            ->setMethods(['offsetGet'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $configArray = [
+            ['IntegrationType', $integrationType]
+        ];
+
+        $mockedCache = $this->createMock(Zend_Cache_Core::class);
+
+        $config = ['directIntegration' => ['enabled' => $directIntegration]];
+
+        $mockedCache->expects($this->once())
+            ->method('load')
+            ->willReturn($config);
+
+        $mockConfig->method('offsetGet')->willReturnMap($configArray);
+        $configLoader = new ConfigLoader(
+            $mockedCache,
+            Shopware()->Container()->get('http_client'),
+            $mockConfig
+        );
+
+        Shopware()->Container()->set('fin_search_unified.config_loader', $configLoader);
+        Shopware()->Container()->set('config', $mockConfig);
+
+        $isDirectIntegration = StaticHelper::checkDirectIntegration();
+
+        // Reset this service directly here as it is being overridden only for this test
+        Shopware()->Container()->reset('fin_search_unified.config_loader');
+        Shopware()->Container()->load('fin_search_unified.config_loader');
+
+        $this->assertSame($directIntegration, $isDirectIntegration);
+
+        /** @var InstallerService $pluginManager */
+        $pluginManager = Shopware()->Container()->get('shopware_plugininstaller.plugin_manager');
+        $plugin = $pluginManager->getPluginByName('FinSearchUnified');
+        $config = $pluginManager->getPluginConfig($plugin);
+
+        $this->assertSame($expectedResult, $config['IntegrationType']);
+    }
+
+    public function directIntegrationProvider()
+    {
+        return [
+            'IntegrationType is API and directIntegration is not enabled' => ['API', 'API', false],
```

Comment:
We have to change the wording a little bit:
* `IntegrationType` -> `Integration type`.
* `directIntegration` -> `DI`.
* `IntegrationType is Direct...` -> `Integration type is DI...`.

We can use the acronym `DI` instead of `Direct Integration`. Customers don't know what `DI` is but since we are the only ones to see it, we will keep it easy for us :wink: